### PR TITLE
Add an :if_not_exists option to create_table

### PR DESCRIPTION
**Pull request by fatkodima**
This PR updates (it has no activity nearly for 2 years) this https://github.com/rails/rails/pull/16376 according to master.
r? @maclover7 @arthurnn

### CODE REVIEW

File: activerecord/test/cases/migration_test.rb [hunk 1]
```diff
@@ -123,6 +123,40 @@ def test_migration_version
     assert_nothing_raised { ActiveRecord::Migrator.run(:up, MIGRATIONS_ROOT + "/version_check", 20131219224947) }
   end
 
+  def test_create_table_raises_if_already_exists
+    connection = Person.connection
+    connection.drop_table :testings, if_exists: true
```
Comment (fatkodima) [inline activerecord/test/cases/migration_test.rb]: Just noticed, whole line should be replaced with `force` for `create_table` and same in the next test

Comment (rafaelfranca) [reply activerecord/test/cases/migration_test.rb]: :+1:

**Review by rafaelfranca — DISMISSED** [review]

**Comment by fatkodima** [comment]
@rafaelfranca updated and rebased

**Comment by fatkodima** [comment]
@rafaelfranca Is there any chance to get also this into 5.2?

**Comment by fatkodima** [comment]
@pixeltrix Updated this pr.
@sebcioz Kudos for pushing this forward!

**Comment by pixeltrix** [comment]
@fatkodima thanks! 👍

**Comment by shootingfly** [comment]
Unbelievable! When you create table, you will drop table! Why keep if_exists default to true?
When I migrate from 5.2 to 6.0, this MR drop all tables in stage environment.

**Comment by pixeltrix** [comment]
@shootingfly I'm unsure what you mean by your comment and I noticed your 👎🏻 on [this comment][1] - that reference to `drop_table` is in a Rails test and not in code that will get executed in an application. Are you saying that you saw all your tables dropped when upgrading from Rails 5.2 to 6.0? If so please open a new issue describing the problem, thanks.

[1]: https://github.com/rails/rails/pull/31382#discussion_r156521270

**Pull request merged**